### ISSUE TITLE
kubernetes/cluster/log-dump.sh appears to be broken on G{C,K}E

### ISSUE THREAD

**Issue by ixdy** [kubernetes/kubernetes#24120]
The end of passing GKE runs shows stuff like
```
Dumping master and node logs to /var/lib/jenkins/jobs/kubernetes-e2e-gke/workspace/_artifacts
Master SSH not supported for gke
ERROR: (gcloud.compute.copy-files) Could not fetch instance:
 - The resource 'projects/k8s-jkns-e2e-gke-ci/zones/us-central1-f/instances/https' was not found
ERROR: (gcloud.compute.copy-files) Could not fetch instance:
 - The resource 'projects/k8s-jkns-e2e-gke-ci/zones/us-central1-f/instances/https' was not found
ERROR: (gcloud.compute.copy-files) Could not fetch instance:
 - The resource 'projects/k8s-jkns-e2e-gke-ci/zones/us-central1-f/instances/https' was not found
```

and there are no node logs being saved. It appears to be working on clusters that failed to fully start up, though.

@spxtr @fejta

**Comment by ixdy** [comment]
`kubernetes-e2e-gke-release-1.2` appears to be affected, too, so probably not due to a code change in Kubernetes.

**Comment by ixdy** [comment]
Might be related to #24056. cc @jlowdermilk

**Comment by jlowdermilk** [comment]
It looks like the latest version of gcloud might broke `detect-node-names` for GKE. [This line](https://github.com/kubernetes/kubernetes/blob/master/cluster/gke/util.sh#L230) is getting the fully-qualified self-link for the node instead of just the instance name.

**Comment by ixdy** [comment]
Now this is broken on GCE, too. From http://ci-test.k8s.io/kubernetes-jenkins/logs/kubernetes-e2e-gce/15524/:

```
Dumping master and node logs to /workspace/_artifacts
Warning: Permanently added '130.211.140.12' (ECDSA) to the list of known hosts.
ERROR: (gcloud.compute.copy-files) Could not fetch instance:
 - The resource 'projects/k8s-jkns-e2e-gce/zones/us-central1-f/instances/https' was not found
ERROR: (gcloud.compute.copy-files) Could not fetch instance:
 - The resource 'projects/k8s-jkns-e2e-gce/zones/us-central1-f/instances/https' was not found
ERROR: (gcloud.compute.copy-files) Could not fetch instance:
 - The resource 'projects/k8s-jkns-e2e-gce/zones/us-central1-f/instances/https' was not found
```

We probably need to apply a similar fix from #24162 to `cluster/gce`.

cc @freehan

**Comment by ixdy** [comment]
I'll work on a fix. I think we can use `--format='value(instance)'` to get something a little less fragile.
